examples/mugen/retrieval/video_clip.py

```diff
@@ -75,26 +78,41 @@ class VideoEncoder(nn.Module):
     Adapted from MUGEN's video encoder
         (https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py)
 
+    Attributes:
+        model (nn.Module): Module extracted from :class:`~torchvision.models.video.S3D`.
+            Code reference: https://github.com/pytorch/vision/blob/main/torchvision/models/video/s3d.py
+        return_node_name (str): Name of the last layer extracted from S3D.
+        out_dim (int): Output dimension of VideoEncoder.
+
     Inputs:
         x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
             Size of ``channel`` dimension must be ``3``.
 
     """
 
-    def __init__(
-        self,
-    ):
+    def __init__(self, num_classes: int = 400) -> None:
         super().__init__()
-        self.model = S3D(400)
-        self.out_dim = self.model.fc.in_channels
-        self.model.fc = nn.Identity()
+        base = S3D(num_classes=num_classes)
+        self.return_node_name = "features"
```

Comment:
I think we can have something more similar with the previous implementation:
```
# Getting input channels from the Conv3d layer
self.out_dim = self.model.classifier[1].in_channels

# Transform the classifier into identity
self.model.classifier = nn.Identity()
```

If we do this way, I think we can let the forward as before (no need for additional avgpool and mean)

Reply:
Thanks for the suggestion. I was on the fence about whether I should:

1. Put assumptions on the layer structure of `classifier` (that the indexing is always `1`) and the existence of attribute `in_channels`

vs.

2. Treat TorchVision's S3D a blackbox with API `create_feature_extractors`, for example, if for some reason `in_channels` becomes invalid it won't affect my callsite as I've been deriving it by passing actual data through the extracted features:

https://github.com/facebookresearch/multimodal/blob/82beb8f726bc1e19c5489f9cf6e30b580e0608a7/examples/mugen/retrieval/video_clip.py#L98-L102

The downside of 2) is that I have to manually attach `Avg_Pool3d` and `torch.mean` in my `forward` pass. 
On the other hand, if the internals of TorchVision S3D remain stable over time, 1) will be the cleaner solution here.

What do you think?

Reply:
Hi Lan, I think since S3D already become a public API in torchvision, it will be likely quite stable (we will be careful to do any change on it). And I think using the `create_feature_extractors` you already assume to use `model.features` in the model.

I what we need to consider here is where this `VideoEncoder` can be generic (using other encoder, not just S3D) or not. For this, I think its better to have `S3DEncoder` that specific to S3D, and make it standard there (like what we did here), then the `VideoEncoder` can accept any `standard` encoder.

Reply:
>I think since S3D already become a public API in torchvision, it will be likely quite stable 
That makes sense to me.

>have S3DEncoder that specific to S3D, and make it standard there (like what we did here), then the VideoEncoder can accept any standard encoder.

What do you mean here by "standard" encoder? So for our `VideoEncoder` to accept any encoder, we will have to request the latter to expose the same attributes (e.g., `classifier`, `in_channels`) as those of TorchVision S3D.

Reply:
Here is a snippet to illustrate my comment:
```
class VideoEncoder(nn.Module):
  def forward(self, x):
    if x.shape[1] != 3:
      ...
    return self.model(x)
    
class S3DEncoder(VideoEncoder):
  def __init__(self):
    ...
```

Reply:
Thanks for the code sample! Since we haven't yet needed to generalize, I will keep it specific to using S3D (with the classifier layer replaced) for now.